### PR TITLE
add test reference for an error message

### LINKED ISSUE #1953: Bad error message for `Optional[list[str] | str]`
repro: 

```python
# from typing import Optional # uncomment for CPython; comment for LPython

NEWLINE      : str       = "\n"
WHITESPACE   : list[str] = [" ", ",", NEWLINE, "\r", "\t"]


def match(pat : str, string : str) -> Optional[list[str] | str]:
    return None
```

produces:

```bash
└─(06:39:53 on brian-lasr ✹ ✭)──> ~/CLionProjects/lpython/src/bin/lpython -I. lasr_lexer_optional_list.py                                             2 ↵ ──(Mon,Jun19)─┘
semantic error: Variable 'list' not declared
 --> lasr_lexer_optional_list.py:7:48
  |
7 | def match(pat : str, string : str) -> Optional[list[str] | str]:
  |                                                ^^^^ 
```

### PR DESCRIPTION
**Pull request by aryangupta701**
fixes #1953 

add a test reference in order to address #1953 issue

### CODE REVIEW

**Comment by aryangupta701** [comment]
any suggestions for the test file's name?

**Comment by kmr-srbh** [comment]
> any suggestions for the test file's name?

Hello @aryangupta701! I think the file can be renamed to say `test_optional.py` since we are testing for the `Optional` type, which is a part of LFortran, but not supported here in LPython.

Also, I am not sure if we need a reference for this test. Let's see what @Shaikh-Ubaid has to say on this.

File: tests/errors/test_optional_list.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+NEWLINE      : str       = "\n"
+WHITESPACE   : list[str] = [" ", ",", NEWLINE, "\r", "\t"]
+
+
```
Comment (Shaikh-Ubaid) [inline tests/errors/test_optional_list.py:4]:
```suggestion
```

File: tests/tests.toml [hunk 2]
```diff
@@ -1383,3 +1383,7 @@ run_with_dbg = true
 [[test]]
 filename = "runtime_errors/test_raise_01.py"
 run_with_dbg = true
+
+[[test]]
+filename = "errors/test_optional_list.py"
```
Comment (Shaikh-Ubaid) [inline tests/tests.toml:1388]: We can rename it to `test_optional.py`. You can add this above line 1350 where you have other error tests listed.

Once you rename, make sure to delete the old reference files.

**Review by Shaikh-Ubaid — COMMENTED** [review]
Two minor comments. Rest looks good. Thanks for this!

**Comment by kmr-srbh** [comment]
I request you to link this PR with the issue so that it gets closed automatically when this is merged. You can add `fixes #1953` in the PR description to link it.

**Review by Shaikh-Ubaid — APPROVED** [review]
It looks good! Thanks! Great work!

**Pull request merged**